mysql-test/t/acl_mdl_scalability.test

```diff
@@ -0,0 +1,264 @@
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+# We rely on g++ to compile c-program
+--source include/not_windows.inc
+# Because of 6K connections, mysqld consumes close to
+# 14G virtual memory
+--source include/big_test.inc
+--source include/not_valgrind.inc
+
+--echo #
+--echo # PS-7143 : Server freezes and reaches max connections due to ACL Cache Lock waits
+--echo #
+
+--echo # Check if system can handle 12k concurrent threads
+--echo # 12K because server creates 6K threads and clients
+--echo # another 6k threads/processes
+
+--echo # There is no easy way to check if the system can
+--echo # handle 12K threads. We compile C-program (need g++)
+--echo # and execute it and check return status
+
+--write_file $MYSQL_TMP_DIR/thread_limit_test.cc
+#include <pthread.h>
```

Comment:
Do we need to use pthread here? Wouldn't std::thread do the job?

Reply:
that can work too